### PR TITLE
[12.0] [ADD] Credit control analysis

### PR DESCRIPTION
**Pull request by qgroulard**
Add a SQL view which summarizes level reached and open balance per partner, policy and currency.
Useful for reporting based on level reached by customers (also displayed in partner form view).

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why new module? I think this can fit perfectly in existing module, as it doesn't require anything extra and it's not something that is changing behavior.

File: account_credit_control/models/res_partner.py
Comment (sbidoul) [inline account_credit_control/models/res_partner.py:52]: This field should either go to credit_control_analysis, or as @pedrobaeza suggests, merge the two modules. Merging them is fine with me.

File: account_credit_control/report/account_credit_control_analysis.py
Comment (sbidoul) [inline account_credit_control/report/account_credit_control_analysis.py]: @qgroulard as you proposed, remove amount_residual_currency, it does not make sense.

File: account_credit_control/report/account_credit_control_analysis.py
Comment (sbidoul) [inline account_credit_control/report/account_credit_control_analysis.py]: Can you add a help text explaining more precisely what this open balance is (eg saying it is the open balance on credit control lines of the same policy).

**Review by sbidoul — APPROVED** [review]
Looks good (code review)

**Review by houssine78 — APPROVED** [review]
LGTM

**Comment by houssine78** [comment]
version increment needed

File: account_credit_control/models/credit_control_line.py
Comment (astirpe) [inline account_credit_control/models/credit_control_line.py]: No need of `readonly=True,` since it's the default behavior for a related field in 12.0.

```suggestion
```

File: account_credit_control/report/account_credit_control_analysis.py
Comment (astirpe) [inline account_credit_control/report/account_credit_control_analysis.py]: No need of string="...", since odoo automatically does the job.

```suggestion
        comodel_name="res.partner", readonly=True
```

File: account_credit_control/report/account_credit_control_analysis.py
Comment (astirpe) [inline account_credit_control/report/account_credit_control_analysis.py]: No need of string="...", since odoo automatically does the job.

```suggestion
        comodel_name="res.currency", readonly=True
```

File: account_credit_control/report/account_credit_control_analysis.py
Comment (astirpe) [inline account_credit_control/report/account_credit_control_analysis.py]: No need of string="...", since odoo automatically does the job.

```suggestion
```

**Review by astirpe — CHANGES_REQUESTED** [review]

**Review by astirpe — APPROVED** [review]

**Review by houssine78 — APPROVED** [review]

File: setup/account_credit_control_analysis/setup.py
Comment (sbidoul) [inline setup/account_credit_control_analysis/setup.py]: @qgroulard these setup.py now refers to an non-existent module. Can you remove it?

After that we can merge.

**Comment by qgroulard** [comment]
@OCA/core-maintainers , This PR is ready to merge

**Pull request merged**
